packages/xdl/src/Project.js

```diff
@@ -1892,6 +1892,40 @@ async function stopWebpackServerAsync(projectRoot) {
   });
 }
 
+export async function bundleWebpackAsync(projectRoot, packagerOpts) {
+  let { exp } = await ProjectUtils.readConfigJsonAsync(projectRoot);
+
+  if (!exp.platforms.includes('web')) {
+    console.log(projectRoot, 'expo', Web.getWebSetupLogs());
+    return;
+  }
+  const mode = packagerOpts.dev ? 'development' : 'production';
+  process.env.BABEL_ENV = mode;
+  process.env.NODE_ENV = mode;
+
+  let config = webpackConfig({
+    projectRoot,
+    excludePolyfill: packagerOpts.excludePolyfill,
+    development: packagerOpts.dev,
+    production: !packagerOpts.dev,
+  });
+  let compiler = webpack(config);
+
+  await new Promise((resolve, reject) =>
+    compiler.run(async (error, stats) => {
+      // TODO: Bacon: account for CI
+      if (error) {
+        // TODO: Bacon: Clean up error messages
+        return reject(error);
+      }
+
+      const { stats: statsPath = 'web-build-stats.json' } = packagerOpts;
+      await JsonFile.writeAsync(path.join(projectRoot, statsPath), stats.toJson());
```

Comment:
I think we get an uncaught promise rejection if this method throws an error (because there's no `await` for this callback function anywhere). You could avoid this by calling `resolve(stats)` here and having a separate `await` after the `await new Promise(...)` call like this https://github.com/expo/expo-cli/pull/407#discussion_r263828107.